cmd/armadactl/cmd/scheduling_control.go

```diff
@@ -0,0 +1,165 @@
+	cmd.Flags().StringSliceP("match-labels", "l", []string{}, "Provide a comma separated list of labels. Queues matching all provided labels will have scheduling paused. Defaults to empty.")
+	cmd.Flags().Bool("inverse", false, "Select all queues which do not match the provided parameters")
+	cmd.Flags().Bool("dry-run", false, "Show selection of queues that will be modified in this operation")
+	return cmd
+}
+
+func uncordonQueues(a *armadactl.App) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "queues",
+		Short: "Resume scheduling for select queues",
+		Long:  "Resume scheduling for select queues. This can be achieved by queue name or by labels.",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return initParams(cmd, a.Params)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			matchQueues, err := cmd.Flags().GetStringSlice("match-queues")
+			if err != nil {
+				return fmt.Errorf("error reading queue selection: %s", err)
+			}
+
+			matchLabels, err := cmd.Flags().GetStringSlice("match-labels")
+			if err != nil {
+				return fmt.Errorf("error reading label selection: %s", err)
+			}
+
+			inverse, err := cmd.Flags().GetBool("inverse")
+			if err != nil {
+				return fmt.Errorf("error reading inverse flag: %s", err)
+			}
+
+			dryRun, err := cmd.Flags().GetBool("dry-run")
+			if err != nil {
+				return fmt.Errorf("error reading dry-run flag: %s", err)
+			}
+
+			if len(matchQueues) == 0 && len(matchLabels) == 0 {
+				return fmt.Errorf("either match-queues or match-labels must be set to narrow down queues to uncordon")
+			}
+
+			return a.UncordonQueues(matchQueues, matchLabels, dryRun, inverse)
+		},
+	}
+	cmd.Flags().StringSliceP("match-queues", "q", []string{}, "Provide a comma separated list of queues you'd like to resume scheduling for. Defaults to empty.")
+	cmd.Flags().StringSliceP("match-labels", "l", []string{}, "Provide a comma separated list of labels. Queues matching all provided labels will have scheduling resumed. Defaults to empty.")
+	cmd.Flags().Bool("inverse", false, "Select all queues which do not match the provided parameters")
+	cmd.Flags().Bool("dry-run", false, "Show selection of queues that will be modified in this operation")
+	return cmd
+}
+
+func cordonQueue(a *armadactl.App) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "queue <queue>",
+		Short: "Pause scheduling for a given queue",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return initParams(cmd, a.Params)
+		},
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			queue := args[0]
+
+			dryRun, err := cmd.Flags().GetBool("dry-run")
+			if err != nil {
+				return fmt.Errorf("error reading dry-run flag: %s", err)
+			}
+
+			return a.CordonQueue(queue, dryRun)
+		},
+	}
+	cmd.Flags().Bool("dry-run", false, "Show selection of queues that will be modified in this operation")
```

Comment:
presumably this is not a selection of queues, but just a single queue?

Reply:
Yep